packages/ember-metal/lib/expand_properties.js

```diff
@@ -37,26 +36,26 @@ var END_WITH_EACH_REGEX = /\.@each$/;
   expansion, and is passed the expansion.
 */
 export default function expandProperties(pattern, callback) {
+  if (typeof pattern !== 'string') {
+    throw new EmberError('A computed property key must be a string');
+  }
+
   if (pattern.indexOf(' ') > -1) {
     throw new EmberError(`Brace expanded properties cannot contain spaces, e.g. 'user.{firstName, lastName}' should be 'user.{firstName,lastName}'`);
   }
 
-  if ('string' === typeof pattern) {
-    var parts = pattern.split(SPLIT_REGEX);
-    var properties = [parts];
+  var parts = pattern.split(SPLIT_REGEX);
+  var properties = [parts];
 
-    parts.forEach((part, index) => {
-      if (part.indexOf(',') >= 0) {
-        properties = duplicateAndReplace(properties, part.split(','), index);
-      }
-    });
+  parts.forEach((part, index) => {
```

Comment:
I know this seems completely unrelated to the point of this PR, but could you use a "normal" for loop here instead?